src/transpose_x_to_y.f90

```diff
@@ -51,10 +59,6 @@ subroutine transpose_x_to_y_real(src, dst, decomp)
      integer :: s1, s2, s3, d1, d2, d3
      integer :: ierror
 
-#ifdef PROFILER
-     if (decomp_profiler_transpose) call decomp_profiler_start("transp_x_y_r")
-#endif
-
```

Comment:
Will this not now miss calls to transpose routines when using the full interface, e.g. in the divergence/Poisson routines in Xcompact3d?

Would it be better to restore the original profiler and add a special case "transp_x_y_memcpy" for the special case when a copy is used instead of communication?

Reply:
I think this is ok. The full interface is not available outside of the library.

Reply:
See https://github.com/2decomp-fft/2decomp-fft/blob/always_profile_transpose/src/decomp_2d.f90#L188

Reply:
It's not? Then these will break https://github.com/xcompact3d/Incompact3d/blob/34d69d2170bc172faafbdf80f697298ad9e89d0a/src/navier.f90#L317

Reply:
oh, I see, my confusion.